src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/SampleSvcClient.java

```diff
@@ -1,24 +1,135 @@
 
-/** Service responsible for making client calls to the Sample service */
-public interface SampleSvcClient {
+/** HTTP RestClient implementation for calls to the Sample service */
+@Component
+public class SampleSvcClient {
+  private static final Logger log = LoggerFactory.getLogger(SampleSvcClient.class);
+
+  @Autowired private AppConfig appConfig;
+
+  @Autowired private RestTemplate restTemplate;
+
+  @Autowired private SampleLinkRepository sampleLinkRepository;
+
+  @Qualifier("sampleRestUtility")
+  @Autowired
+  private RestUtility restUtility;
+
+  @Autowired private SurveySvcClient surveyService;
 
   /**
    * Request the delivery of sample units from the Sample Service via a message queue.
    *
    * @param exercise the Collection Exercise for which to request sample units.
    * @return the total number of sample units in the collection exercise.
    */
-  SampleUnitsRequestDTO requestSampleUnits(CollectionExercise exercise) throws CTPException;
+  @Retryable(
+      value = {RestClientException.class},
+      maxAttemptsExpression = "#{${retries.maxAttempts}}",
+      backoff = @Backoff(delayExpression = "#{${retries.backoff}}"))
+  public SampleUnitsRequestDTO requestSampleUnits(CollectionExercise exercise) throws CTPException {
+
+    UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getSampleSvc().getRequestSampleUnitsPath(), null, exercise);
+
+    List<SampleLink> sampleLinks =
+        sampleLinkRepository.findByCollectionExerciseId(exercise.getId());
+    List<UUID> sampleSummaryUUIDList = new ArrayList<>();
+    for (SampleLink samplelink : sampleLinks) {
+      sampleSummaryUUIDList.add(samplelink.getSampleSummaryId());
+    }
+
+    SurveyDTO surveyDto = this.surveyService.findSurvey(exercise.getSurveyId());
+
+    if (surveyDto == null) {
+      throw new CTPException(
+          CTPException.Fault.BAD_REQUEST,
+          String.format(
+              "Invalid survey %s for collection exercise %s",
+              exercise.getSurveyId(), exercise.getId()));
+    } else {
+      CollectionExerciseJobCreationRequestDTO requestDTO =
+          new CollectionExerciseJobCreationRequestDTO();
+      requestDTO.setCollectionExerciseId(exercise.getId());
+      requestDTO.setSurveyRef(surveyDto.getSurveyRef());
+      requestDTO.setExerciseDateTime(exercise.getScheduledStartDateTime());
+      requestDTO.setSampleSummaryUUIDList(sampleSummaryUUIDList);
+
+      HttpEntity<CollectionExerciseJobCreationRequestDTO> httpEntity =
+          restUtility.createHttpEntity(requestDTO);
+
+      log.with("collection_exercise_id", exercise.getId().toString())
+          .debug("Requesting sample unit for collection exercise");
+      ResponseEntity<SampleUnitsRequestDTO> responseEntity =
+          restTemplate.exchange(
+              uriComponents.toUri(), HttpMethod.POST, httpEntity, SampleUnitsRequestDTO.class);
+      return responseEntity.getBody();
+    }
+  }
+
+  @Retryable(
```

Comment:
The `@Retryable` annotation should be used very sparingly. We can't have users waiting 25 seconds for a response.